### PR TITLE
chore: jsdoc stuff

### PR DESCRIPTION
**Pull request by cristiand391**
Adds jsdoc comments in bulk2 module and other small ones.

Also makes query/ingest job polling methods to include an error when emitting `failure`.

### CODE REVIEW

File: src/api/bulk.ts [hunk 1]
```diff
@@ -1233,14 +1284,17 @@ export class QueryJobV2<S extends Schema> extends EventEmitter {
...
     const url = `${this.#connection.instanceUrl}/services/data/v${
       this.#connection.version
     }/jobs/query/${getJobIdOrError(this.jobInfo)}/results`;
 
     return this.locator ? `${url}?locator=${this.locator}` : url;
   }
 
+  /**
+   * Get the results for a query job.
+   */
```
Comment (shetzel) [inline src/api/bulk.ts:1297]: Add `returns` jsdoc

File: src/api/bulk.ts [hunk 2]
```diff
@@ -1160,6 +1196,19 @@ export class QueryJobV2<S extends Schema> extends EventEmitter {
...
+   * This method will only throw after a timeout. To capture a
+   * job failure while polling you must set a listener for the
+   * `failed` event before calling it:
+   *
+   * job.on('failed', (err) => console.error(err))
+   * await job.poll()
+   *
+   * @param interval Polling interval in milliseconds
+   * @param timeout Polling timeout in milliseconds
+   */
```
Comment (shetzel) [inline src/api/bulk.ts:1211]: Add `returns`?

File: src/api/bulk.ts [hunk 3]
```diff
@@ -1405,6 +1469,19 @@ export class IngestJobV2<
...
```
Comment (shetzel) [inline src/api/bulk.ts:1484]: Add `returns`?

File: src/connection.ts [hunk 4]
```diff
@@ -470,6 +470,9 @@ export class Connection<S extends Schema = Schema> extends EventEmitter {
    * Authorize the connection using OAuth2 flow.
    * Typically, just pass the code returned from authorization server in the first argument to complete authorization.
    * If you want to authorize with grant types other than `authorization_code`, you can also pass params object with the grant type.
+   *
+   * @returns {Promise<AuthInfo>} An object that contains the user ID, org ID and identity URL.
```
Comment (shetzel) [inline src/connection.ts:474]: This is `Promise<UserInfo>` right?

File: src/api/bulk.ts [hunk 5]
```diff
@@ -1024,7 +1024,24 @@ export class BulkV2<S extends Schema> {
...
+   * });
+   *
+   * // create the job in the org
+   * await job.open()
+   *
+   * // upload data
+   * await job.uploadData(csvFile)
+   *
+   * // finished uploading data, mark it as ready for processing
+   * await job.close()
```
Comment (shetzel) [inline src/api/bulk.ts:1044]: Add params and returns?

**Review by shetzel — APPROVED** [review]
I have some non-blocking enhancements to the jsdoc

File: src/connection.ts (hunk 4 above)
Comment (cristiand391) [reply src/connection.ts:474]: in jsdoc types are inside curly brackets, we use the same syntax in sfdx-core
https://github.com/search?q=repo%3Aforcedotcom%2Fsfdx-core+%2F%40returns+%5C%7BPromise%2F&type=code

https://jsdoc.app/tags-returns.html

**Comment by cristiand391** [comment]
@shetzel I created W-14259530 to track fixing the flaky tests 👍🏼

**Pull request merged**